pkg/converter/converter.go

```diff
@@ -404,8 +404,35 @@ func convertParameters(context *configConvertContext, environment manifest.Envir
 			}
 			parameters[newName] = &listParam.ListParameter{Values: valueSlice}
 		} else if regex.IsEnvVariable(value) {
-			envVarName := regex.TrimToEnvVariableName(value)
-			parameters[newName] = envParam.New(envVarName)
+			var refs []parameter.ParameterReference
+			envParams := extractEnvParameters(value)
+
+			// convert found environment variables to compound parameter referencing these variables
+			valueWithPlaceHolders := value
+			for _, p := range envParams {
+				p := p // avoid implicit memory aliasing
+
+				parameterName := fmt.Sprintf("__ENV_%s__", p.Name)
+				parameters[parameterName] = &p
+				valueWithPlaceHolders = strings.ReplaceAll(strings.ReplaceAll(valueWithPlaceHolders, p.Name, parameterName), ".Env", "")
+				// create references
+				refs = append(refs, parameter.ParameterReference{
+					Config: coordinate.Coordinate{
+						Project:  config.GetProject(),
+						Type:     config.GetType(),
+						ConfigId: config.GetId(),
+					},
+					Property: parameterName,
+				})
+			}
+
+			// convert to compound parameter
+			c, err := compound.New(newName, valueWithPlaceHolders, refs)
```

Comment:
For the case of a single variable that is not a compound, couldn't we still exit before adding the compound iff there's only a single param and the `valueWithPlaceholders` is exactly equal to `{{ .Env.[p.Name] }}`?

Reply:
yes, but why?

Reply:
I think creating a compound where none is needed at all can be confusing to users. 
And if it's an easy fix to not do this with a 3 line early exit if in the right place, I don't see why we should why we should do this only half-way and produce unnecessary complexity.